lib/rucio/core/did_meta_plugins/filter_engine.py

```diff
@@ -235,72 +214,121 @@ def _translate_filters(self, model_class, strict_coerce=True):
-                or_group.append(list(filter))
+            for _filter in additional_filters:
+                or_group.append(list(_filter))
 
         or_expressions = []
         for or_group in self._filters:
             and_expressions = []
             for and_group in or_group:
                 key, oper, value = and_group
-                if isinstance(value, str) and any([char in value for char in ['*', '%']]):  # wildcards
-                    if value in ('*', '%', u'*', u'%'):                                     # match wildcard exactly == no filtering on key
-                        continue
-                    else:                                                                   # partial match with wildcard == like || notlike
-                        if session.bind.dialect.name == 'postgresql':
+                if isinstance(key, sqlalchemy.orm.attributes.InstrumentedAttribute):                # -> this key filters on a table column.
+                    if isinstance(value, str) and any([char in value for char in ['*', '%']]):      # wildcards
+                        if value in ('*', '%', u'*', u'%'):                                         # match wildcard exactly == no filtering on key
+                            continue
+                        else:                                                                       # partial match with wildcard == like || notlike
                             if oper == operator.eq:
                                 expression = key.like(value.replace('*', '%').replace('_', '\_'), escape='\\')     # NOQA: W605
                             elif oper == operator.ne:
                                 expression = key.notlike(value.replace('*', '%').replace('_', '\_'), escape='\\')  # NOQA: W605
+                    else:
+                        expression = oper(key, value)
+                    if oper == operator.ne:                                                         # set .ne operator to include NULLs.
+                        expression = or_(expression, key.is_(None))
+                elif json_column:                                                                   # -> this key filters on the content of a json column
+                    if session.bind.dialect.name == 'oracle':
+                        # Compiling for Oracle yields a UnsupportedCompilationError so fall back to support for equality only.
+                        # TODO
+                        if oper != operator.eq:
+                            raise exception.FilterEngineGenericError("Oracle implementation does not support this operator.")
+                        if isinstance(value, str) and any([char in value for char in ['*', '%']]):
+                            raise exception.FilterEngineGenericError("Oracle implementation does not support wildcards.")
+                        expression = text("json_exists(meta,'$?(@.{} == \"{}\")')".format(key, value))
```

Comment:
yet to implement this for Oracle - requires a more bespoke filter expression construction